K2Bridge/KustoDAL/KustoResponseParser.cs

```diff
@@ -189,10 +194,20 @@ private void ReportQueryExecutionMetrics(KustoResponseDataSet kustoResponseDataS
                 {
                     foreach (DataRow row in parsedKustoResponse[AggregationTableName].TableData.Rows)
                     {
-                        IBucket bucket = createBucketFromDataRow(row);
-                        response.AddBucketToAggregation(bucket);
+                        Bucket bucket = createBucketFromDataRow(row);
+                        if (bucket != null)
+                        {
+                            response.AddBucketToAggregation(bucket);
+                        }
                     }
                 }
+
+                // For Range aggregations, the calculated total hits is wrong, so we have an additional column with the expected count
+                if (parsedKustoResponse[HitsTotalTableName] != null)
```

Comment:
so this exists only for range query?

Reply:
Currently, for any aggregation... As an optimization we could emit the `hitsTotal` query for Range only, yes.

https://github.com/microsoft/K2Bridge/pull/79/files/70e32a243b509ee13d457f6beca27f39c08e3f6d#diff-6ee10d9b710b3a5b20475e9968163ec8723bd8e6319f9aed7ed2b23f80695767R72